enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,597 @@
+provides a message indicating to the cluster admin that they must delete the service.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: custom
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  domain: mycluster.com
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: External
+      providerParameters:
+        type: AWS
+        aws:
+          type: Classic
+          subnets:
+            - subnetA
+            - subnetB
+            - subnetC
+status:
+  availableReplicas: 2
+  conditions:
+  - message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  - message: "The IngressController subnets were changed from [\"\"] to [\"subnetA\", \"subnetB\", \"subnetC\"]. To effectuate this change, you must delete the service: `oc -n openshift-ingress delete svc/custom-default`; the service load-balancer will then be deprovisioned and a new one created. This will most likely cause the new load-balancer to have a different host name and IP address from the old one's. Alternatively, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"type\":\"LoadBalancerService\",\"loadBalancer\":{\"providerParameters\":{\"type\":\AWS\",\"aws\":{\"type\":\"Classic\",\"subnets\":[]}}}}}}'"
+    reason: OperandsProgressing
+    status: "True"
+    type: LoadBalancerProgressing
+```
+
+We have intentionally added this requirement for manual deletion of the
+service for the following reasons:
+
+##### Reason #1: To mitigate risks associated with cluster admins providing an invalid annotation value.
+
+While the AWS Cloud Controller Manager (CCM) handles invalid annotations by
+producing errors and events, it is difficult for the Ingress Operator to
+decipher these CCM events _after_ the load balancer has been provisioned.
+Therefore, the Ingress Operator can't indicate to a cluster admin that an
+IngressController's load balancer is in a malfunctioning state (i.e., the
+service is not getting reconciled) while the service is already provisioned.
+
+When the IngressController is first created and the LoadBalancer-type
+service is not yet provisioned, these same invalid `Subnets` values will
+prevent the load balancer from being provisioned. The existing Ingress
+Operator logic will clearly indicate to the cluster admin that the load balancer
+failed to provision via the `LoadBalancerReady` status condition. In addition,
+the `LoadBalancerReady` condition will include the CCM event logs that indicate
+to the cluster admin that the `Subnets` value is invalid (see [Support Procedures](#support-procedures)
+for examples). Therefore, requiring manual service deletion provides a way
+for the Ingress Operator to produce a clear signal to the cluster admin
+that `Subnets` is invalid.
```

Comment:
I actually think this is useful context and would encourage you to keep it. I'm coming to this conversation fresh today and would probably be asking many more questions if this section/paragraph were not here.

In general for enhancements we have to assume a minimal base knowledge so that they are accessible to everyone